### DISCUSSION TITLE
能否做到像processon那样可以在两个节点之间任意连线？

### DISCUSSION THREAD

**Discussion by happy-little-one** [antvis/X6#2253] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/13190639/174831429-ef798eac-5bd4-4da3-af7b-192fa013d496.png)

process的交互逻辑是，可以在节点边缘处任意位置开始划线，然后可以链接到另一节点边缘的任务位置把它们连起来，不依赖桩点。

x6所有的连线都是基于桩点的，我能想到的是动态添加桩点，但是这样还要在删除线的时候再删除桩点，感觉不是特别好做，不知有没有更好的方案？谢谢！

**Comment by NewByVector** [comment]
x6 也是支持在节点之间连线的，不过默认线都会重合在一起，可以使用 [NodeAnchor](https://x6.antv.vision/zh/docs/api/registry/node-anchor/#gatsby-focus-wrapper) 将它们区分开来。

**Reply by happy-little-one** [reply]
感觉这个没法做到，它的目的感觉设置某种模式，而不是精确设置位置。

**Comment by happy-little-one** [comment, marked as answer]
找到办法了，可以通过edge的tools vertices和segment实现

**Reply by wenhaoxu1** [reply]
使用 edge的tools vertices和segment也是让线分开吧,能精准设置线的source跟target的位置吗

**Reply by lanzaiyang** [reply]
大佬，方便分享一下是怎么实现得吗

**Reply by fireflyshen** [reply]
> 大佬，方便分享一下是怎么实现得吗

这种一看就是锚点呀，他说的是实现processon那种连线的模式

**Reply by lanzaiyang** [reply]
> > 大佬，方便分享一下是怎么实现得吗
> 
> 这种一看就是锚点呀，他说的是实现processon那种连线的模式

是的，processon的那种连线模式，基于x6可以实现的吗，你有思路吗

**Reply by fireflyshen** [reply]
> > > 大佬，方便分享一下是怎么实现得吗
> > 
> > 
> > 这种一看就是锚点呀，他说的是实现processon那种连线的模式
> 
> 是的，processon的那种连线模式，基于x6可以实现的吗，你有思路吗

我刚看了一下，如果你没个节点都连一条线的话，那很简单，有个magnet属性设置为true就行了，这个属性一开整个节点变成连接桩，如果是processon那种更像是解析路径当触发mousedown事件的时候动态生成连接桩，连线，麻烦一点的可以解析路径，方便一点的point-event:stroke（好像是这个属性），这样只有路径才会触发你的鼠标事件，监听他动态创建连接桩就行。。。

**Reply by TLovers** [reply]
> > > > 大佬，方便分享一下是怎么实现得吗
> > > 
> > > 
> > > 这种一看就是锚点呀，他说的是实现processon那种连线的模式
> > 
> > 
> > 是的，processon的那种连线模式，基于x6可以实现的吗，你有思路吗
> 
> 我刚看了一下，如果你没个节点都连一条线的话，那很简单，有个magnet属性设置为true就行了，这个属性一开整个节点变成连接桩，如果是processon那种更像是解析路径当触发mousedown事件的时候动态生成连接桩，连线，麻烦一点的可以解析路径，方便一点的point-event:stroke（好像是这个属性），这样只有路径才会触发你的鼠标事件，监听他动态创建连接桩就行。。。

process是两点之间可以随意连线，x6设置磁性所有的线都会重合到一起
---
路径才会触发你的鼠标事件，监听他动态创建连接桩就行   这句哈如何理解啊，大佬

**Comment by TLovers** [comment]
> 找到办法了，可以通过edge的tools vertices和segment实现

大佬怎么弄的啊，有代码实现思路吗
